### PR TITLE
Warn instead of fatal if deleted host is not found in k8s.

### PR DESCRIPTION
**Pull request by moelsayed**
https://github.com/rancher/rke/issues/70

### CODE REVIEW

File: hosts/hosts.go [hunk 1]
```diff
@@ -56,13 +57,19 @@ func (h *Host) CleanUp() error {
 
 func DeleteNode(toDeleteHost *Host, kubeClient *kubernetes.Clientset) error {
 	logrus.Infof("[hosts] Cordoning host [%s]", toDeleteHost.Address)
-	err := k8s.CordonUncordon(kubeClient, toDeleteHost.HostnameOverride, true)
-	if err != nil {
+	if _, err := k8s.GetNode(kubeClient, toDeleteHost.HostnameOverride); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		} else {
+			logrus.Warnf("[hosts] Host [%s] is not a Kubernetes node")
```
Comment (galal-hussein) [inline hosts/hosts.go]: can you give the warning some context, like `can't cordon the host because ...`

File: hosts/hosts.go [hunk 2]
```diff
@@ -56,13 +57,19 @@ func (h *Host) CleanUp() error {
 
 func DeleteNode(toDeleteHost *Host, kubeClient *kubernetes.Clientset) error {
 	logrus.Infof("[hosts] Cordoning host [%s]", toDeleteHost.Address)
-	err := k8s.CordonUncordon(kubeClient, toDeleteHost.HostnameOverride, true)
-	if err != nil {
+	if _, err := k8s.GetNode(kubeClient, toDeleteHost.HostnameOverride); err != nil {
+		if !apierrors.IsNotFound(err) {
```
Comment (alena1108) [inline hosts/hosts.go]: Start with `if apierrors.isNotFound`, and if met, log a messgae `Can't find k8s node by name []`

File: hosts/hosts.go [hunk 3]
```diff
@@ -56,13 +57,19 @@ func (h *Host) CleanUp() error {
...
 func DeleteNode(toDeleteHost *Host, kubeClient *kubernetes.Clientset) error {
 	logrus.Infof("[hosts] Cordoning host [%s]", toDeleteHost.Address)
-	err := k8s.CordonUncordon(kubeClient, toDeleteHost.HostnameOverride, true)
-	if err != nil {
+	if _, err := k8s.GetNode(kubeClient, toDeleteHost.HostnameOverride); err != nil {
+		if apierrors.IsNotFound(err) {
+			logrus.Warnf("[hosts] Can't find node by name [%s]")
+			return nil
+		} else {
+			return err
```
Comment (alena1108) [inline hosts/hosts.go]: I don't think there is a need for `else`. Just return err

File: hosts/hosts.go (hunk 1 above)
Comment (galal-hussein) [inline hosts/hosts.go]: specify the host name for %s

**Pull request merged**